static/js/ducks/sources.js

```diff
@@ -4,14 +4,16 @@ import store from '../store';
 
 export const FETCH_SOURCES = 'skyportal/FETCH_SOURCES';
 export const FETCH_SOURCES_OK = 'skyportal/FETCH_SOURCES_OK';
+export const FETCH_SOURCES_FAIL = 'skyportal/FETCH_SOURCES_FAIL';
 
 
-export function fetchSources(page=1) {
-  return API.GET(`/api/sources?page=${page}`, FETCH_SOURCES);
-}
-
-export function submitSourceFilterParams(formData) {
-  return API.POST(`/api/sources/filter`, FETCH_SOURCES, formData);
+export function fetchSources(filterParams={}) {
+  if (!Object.keys(filterParams).includes("pageNumber")) {
+    filterParams.pageNumber = 1;
+  }
+  const params = new URLSearchParams(filterParams);
```

Comment:
This looks good to me, similar to what is suggested [here](https://github.com/github/fetch/issues/256#issuecomment-170228674).

I'm going to merge, but how about we add this as a general feature of `API.GET`, so that we can utilize it in other places too?

I.e., `GET(endpoint, actionType, queryParams)`.

Reply:
Good call! I'll do this tomorrow.